File: app/src/main/java/com/arcao/opencaching4locus/ui/dashboard/DashboardActivity.java
package com.arcao.opencaching4locus.ui.dashboard;

import android.app.Activity;
import android.os.Bundle;
import android.widget.Toast;
import com.arcao.opencaching4locus.App;
import com.arcao.opencaching4locus.ui.base.BaseActivity;
import dagger.Binds;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import javax.inject.Inject;

/**
 * Created by Arcao on 30.08.2017.
 */

public class DashboardActivity extends BaseActivity {
  @Inject App app;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    Toast.makeText(app, "test", Toast.LENGTH_LONG).show();
    finish();
  }

  // =============== Dagger 2 ===============
  @dagger.Subcomponent
  interface Component extends AndroidInjector<DashboardActivity> {

    @dagger.Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<DashboardActivity> {}
  }

  @dagger.Module(subcomponents = Component.class)
  abstract class Module {
    @Binds
    @IntoMap
    @ActivityKey(DashboardActivity.class)
    abstract AndroidInjector.Factory<? extends Activity> bind(Component.Builder builder);
  }
}

File: app/src/main/kotlin/com/arcao/opencaching4locus/ui/dashboard/DashboardActivity.kt
package com.arcao.opencaching4locus.ui.dashboard

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import com.arcao.opencaching4locus.R
import com.arcao.opencaching4locus.databinding.ActivityDashboardBinding
import com.arcao.opencaching4locus.ui.base.BaseActivity
import javax.inject.Inject

class DashboardActivity : BaseActivity() {
    @Inject internal lateinit var viewModelFactory: ViewModelProvider.Factory
    private lateinit var viewModel: DashboardViewModel
    private lateinit var binding: ActivityDashboardBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        viewModel = ViewModelProviders.of(this, viewModelFactory).get(DashboardViewModel::class.java)
        viewModel.intent = intent
        viewModel.startActivityLiveData.observe(this, Observer {
            startActivityForResult(it, 0)
        })

        binding = DataBindingUtil.setContentView(this, R.layout.activity_dashboard)
        binding.viewModel = viewModel
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (resultCode == RESULT_OK) {
            finish()
        }
    }
}
